Basic GNU/Linux compilation instructions (with a focus on Debian 10)

polarisfm opened the pull request:
I've added some basic GNU/Linux compilation instructions. This was done on PureOS Amber, which is a version of Debian 10 which removes some non-free components to meet the Free Software Foundation's guidelines for what's considered a free distribution. The packages relevant to compiling e621 downloader were not modified from Debian 10's packages.

This should hold true for other GNU/Linux systems (minus instructions for installing rustc and cargo).

Feel free to ask any questions or make needed edits to better fit your preferred documentation style.

In case it does not go without saying, these contributions are licensed under the Apache License 2.0.

McSib commented:
Off of reading the changes you've made, the only correction I would say would be the rust version. The latest version of rustc is 1.45.0, is this possibly different on Linux systems?

Also, I would clarify that when building the project, if the user has the intent to build with release code, to type in `cargo build` with the addition of `--release` at the end. 

Beyond that, it looks good! If these changes can be made, it should be good to merge with the active branch and help others set up the project faster.

polarisfm commented:
The reason rustc and cargo are out-of-date is because most LTS distributions don't push the newest versions of packages without extensive testing. Libraries are also usually frozen at specific versions.

I've added instructions for people who choose to use the versions of rustc and cargo from Debian's repositories ("Method 1"), as well as directions to people who wish to use the latest version of rust via rustup ("Method 2").

Method 1 comes with a disclaimer saying that builds may fail because of the often out-of-date packages, and to try Method 2 before filing an issue.

I've also added a bit telling people where the binaries are located after compilation and how to make them executable if they have issues.

Sorry for the delayed response, I was attending to some family things.

akarras commented:
rustup is the preferred method for installing rustc, as you said it can be outdated, but rustup will also configure paths appropriately and isn't unique to Debian. I would reckon that preference should also carry over to this repo.

Also, instead of cloning the repo and building a release, it's likely much easier from a user perspective to simply `cargo install --git https://github.com/McSib/e621_downloader --branch active`. Assuming you used rustup, this should produce a release binary in `.cargo/bin` that will be executable on the user's path on every platform.

Additionally, getting the crate published on crates.io would make this a bit easier as well as users would be able to simply `cargo install e621_downloader`

Now I need to figure out what e621 is, I just had a friend point me to this repo asking for help. :)

McSib commented:
> rustup is the preferred method for installing rustc, as you said it can be outdated, but rustup will also configure paths appropriately and isn't unique to Debian. I would reckon that preference should also carry over to this repo.
> 
> Also, instead of cloning the repo and building a release, it's likely much easier from a user perspective to simply `cargo install --git https://github.com/McSib/e621_downloader --branch active`. Assuming you used rustup, this should produce a release binary in `.cargo/bin` that will be executable on the user's path on every platform.
> 
> Additionally, getting the crate published on crates.io would make this a bit easier as well as users would be able to simply `cargo install e621_downloader`
> 
> Now I need to figure out what e621 is, I just had a friend point me to this repo asking for help. :)

E621 is a mature image board replacement for the image board Sidechan. A general audience image board, e926 (formerly e961) complements this site. E621 runs off of the Ouroboros platform, a danbooru styled software specifically designed for the site.

McSib commented:
It also carries a few similarities to 4chan in its layout. But generally it's a not safe for work furry image board platform, with their safe for work platform being e926. I would recommend if you're not familiar with the furry community or imageboard sites like this to go through e926. All it does is filter out the explicit posts and only shows the safe post. That should be efficient enough for you to check out their api, and add on to it if you feel the need to.

McSib commented:
I'm going to formally close this pull request as I made a commit a couple of months back that includes much of the information needed to install/compile this program onto Windows and Linux base systems. If anything needs to be added, altered or filled out, update the information inside the new README.md and propose a new pull request.

The commit that adds the new information mentioned above: https://github.com/McSib/e621_downloader/commit/bf923e6b8d78a892e42830fa448d83322ac52566
